paper-api/src/main/java/org/bukkit/damage/DamageEffect.java

```diff
@@ -6,39 +6,41 @@
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 
+// Paper start
 /**
  * Represents a type of effect that occurs when damage is inflicted. Currently,
  * effects only determine the sound that plays.
  */
 @ApiStatus.Experimental
-public interface DamageEffect {
+public enum DamageEffect {
```

Comment:
Using an enum instead of an interface was the first idea that came to my mind. I know that it comes with the mentioned problems.

Maybe use the id of the damage effect as String instead of the Bukkit Interface for DamageTypeRegistryEntry?

Reply:
Agree with that sentiment ^
Making this an enum is not something I'd like to see.
We'll end up with *some* version of unsafe values at bootstrap time down the line anyway so, *yea*.

Alternatively, we can *attempt* to predict that damage effects *might* be a registry value at some point?
Which would mean we *could* just move this to keys instead of the actual resolved value.
That would need a generated key class tho.
I'd personally be semi in favour of that approach as it means we won't have to break API there if that ever changes and it does get rid of the bootstrap issue. Will poke MM about it tho for his ideas.

Reply:
Maybe [this](https://github.com/PaperMC/Paper/pull/11783/commits/5570b7bac8cece5062d35f79d2a9fab2c7716b9b) could be a solution.

Instead of having an extra class (StaticUnsafeValues), bukkits UnsafeValues interface could have a static getter.

Reply:
I want a separate type called something like “APIBridge” that uses the service loader API for stuff like this, so we can move actual API out of UnsafeValues. UnsafeValues should be for actual “unsafe” api, not just a place to store “bridge” methods. 

But either way, yeah, this shouldn’t be changed back to an enum.